src/js/components/FloatingElement.jsx

```diff
@@ -422,12 +422,16 @@ export class FloatingElement extends Component {
     }
 
     componentRendered() {
-        // As soon as possible, we need to re-calculate our position
-        this.validatePositioningScheduled = true;
-        window.requestAnimationFrame(() => {
-            this.validatePositioningScheduled = false;
+        if (this.props.positionImmediate) {
```

Comment:
I didn't mean keep it around all the time, just when you need it create it with opacity 0, then once it's in the right place, "one frame" set it to 1, so you don't see it appear for a "frame" at 0,0 then blink into the right place.

Or, we just set the initial position as -9999, -9999, see how that goes.

I'm just throwing out ideas to fix the original problem without having to block everything for a synchronous dom validation